### PR TITLE
refactor: initialize the user auth state synchronously

### PR DESCRIPTION
**Pull request by petebacondarwin**
We can now initialize the user state synchronously, which means that
we can remove the checks for whether it has been done or not in each
of the user auth functions.

### CODE REVIEW

**Comment by threepointone** [comment]
Hope this isn't urgent, I'd love to review this before it lands

**Comment by petebacondarwin** [comment]
No not urgent but it is not as scary as you may think.

**Review by JacobMGEvans — COMMENTED** [review]
Interesting! Since things are sync now, you can remove the throw checks and the entire mocking of users and directly test the functions handling the user, i.e.  `reinitialise`,
  `USER_AUTH_CONFIG_FILE`, `writeAuthConfigFile`

File: packages/wrangler/src/user.tsx [hunk 1]
```diff
@@ -311,60 +328,54 @@ const TOKEN_URL = "https://dash.cloudflare.com/oauth2/token";
...
+let LocalState = getAuthState();
 
-// we do this because we have some async stuff
-// TODO: this should just happen in the top level
-// and we should figure out how to do top level await
-export async function initialise(): Promise<void> {
+/**
+ * Compute the current auth state.
+ */
+function getAuthState(): State {
```
Comment (threepointone) [inline packages/wrangler/src/user.tsx:336]: "state" isn't descriptive here, especially becase LocalState contains more stuff. How about `getAuthTokens()`?

File: packages/wrangler/src/user.tsx [hunk 2]
```diff
@@ -311,60 +328,54 @@ const TOKEN_URL = "https://dash.cloudflare.com/oauth2/token";
...
-function throwIfNotInitialised() {
-  if (initialised === false) {
-    throw new Error(
-      "did you forget to call initialise() from the user module?"
-    );
-  }
+/**
+ * Run the initialisation of the auth state, in the case that something changed.
+ */
+export function reinitialise(): void {
```
Comment (threepointone) [inline packages/wrangler/src/user.tsx:374]: Would you like to take this opportunity to rename this function too? It was never clear even before. Got any ideas?

**Review by threepointone — APPROVED** [review]
Looks great! Don't forget to run it a couple of times locally before landing.

File: packages/wrangler/src/user.tsx (hunk 2 above)
Comment (petebacondarwin) [reply packages/wrangler/src/user.tsx:374]: Yeah - I changed it from `initialise()` to `reinitialise()` but I would be up for an even more description name... perhaps `reinitialiseAuth()`?

File: packages/wrangler/src/user.tsx (hunk 1 above)
Comment (petebacondarwin) [reply packages/wrangler/src/user.tsx:336]: I would quite like to go a step further and split apart the OAuth state processing from the external auth token state. But I thought that would be too large a change in this PR.

The OAuth state fields that are stored in `State` really an internal implementation thing which code outside of `user.tsx` should not care about.

**Pull request merged**